.github/workflows/publish.yml

```diff
@@ -25,4 +28,34 @@ jobs:
     secrets: inherit
     with:
       publish: true
-      repoName: finality-gadget
\ No newline at end of file
+      repoName: finality-gadget
+      
+  test_finality_provider:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Clone finality-provider repository
+        run: |
+          current_hash=$(git rev-parse HEAD)
+          remote_url=$(git remote get-url origin)
+          if [[ $remote_url == https://* ]]; then
+              # For HTTPS URLs
+              repo_path=$(echo $remote_url | sed -E 's|https://github.com/(.+)(\.git)*|\1|')
+          else
+              # For SSH URLs
+              repo_path=$(echo $remote_url | sed -E 's/.*:(.+)\.git/\1/')
+          fi
+          fp_version=$(go list -m github.com/$repo_path@$current_hash)
```

Comment:
The goal here seems to get the version of the current FP repository. Can we achieve this by `git rev-parse HEAD`?

Reply:
> The goal here seems to get the version of the current FP repository. Can we achieve this by `git rev-parse HEAD`?

line 41. However it seems only works for push action. For pull request, should I use HEAD~ instead?

Reply:
Looks good by now  :)
https://github.com/babylonlabs-io/finality-gadget/actions/runs/11772186929/job/32787208311